app/services/spotlight/exhibit_import_export_service.rb

```diff
@@ -0,0 +1,379 @@
+        page[:upload] = serialize_featured_image(page[:upload_id]) if page[:upload_id]
+        page.delete(:upload_id)
+      end
+
+      json
+    end
+
+    def serialize_featured_image(id)
+      image = Spotlight::FeaturedImage.find(id)
+      file = image.image.file
+      if file
+        img = {
+          image: {
+            filename: file.identifier, content_type: file.content_type, content: Base64.encode64(file.read)
+          }
+        }
+      end
+
+      image.as_json(except: %i[id image]).merge(img || {}).deep_symbolize_keys
+    end
+
+    def attach_attachments(json)
+      return json unless json[:attachments]
+
+      json[:attachments].each do |attachment|
+        a = exhibit.attachments.find(attachment[:id])
+        file = a.file.file
+
+        attachment[:file] = {
+          filename: file.filename,
+          content_type: file.content_type,
+          content: Base64.encode64(file.read)
+        }
+
+        attachment.delete(:id)
+      end
+
+      json
+    end
+
+    def add_feature_page_hierarchy(json)
+      return json unless json[:feature_pages]
+
+      page_id_map = json[:feature_pages].map { |x| [x[:id], x[:slug]] }.to_h
+
+      json[:feature_pages].each do |page|
+        page.delete(:id)
+        next unless page[:parent_page_id]
+
+        page[:parent_page_slug] = page_id_map[page[:parent_page_id]]
+        page.delete(:parent_page_id)
+      end
+
+      json
+    end
+
+    def add_page_content(json)
+      (json[:feature_pages] || []).each do |page|
+        page[:content] = exhibit.feature_pages.find_by(slug: page[:slug]).read_attribute(:content)
+      end
+
+      (json[:about_pages] || []).each do |page|
+        page[:content] = exhibit.about_pages.find_by(slug: page[:slug]).read_attribute(:content)
+      end
+
+      json[:home_page][:content] = exhibit.home_page.read_attribute(:content) if json[:home_page]
+
+      json
+    end
+
+    def stringify_classes(json)
+      return json unless json[:solr_document_sidecars]
+
+      json[:solr_document_sidecars].each do |obj|
+        obj[:document_type] = obj[:document_type].to_s
+      end
+
+      json
+    end
+
+    def raw_json(_input = nil)
+      exhibit.as_json(
+        {
+          except: %i[id slug site_id],
+          include: {}.merge(
+            if_include?(:config,
+                        main_navigations: {
+                          except: %i[id exhibit_id]
+                        },
+                        contact_emails: {
+                          except: %i[id exhibit_id confirmation_token]
+                        },
+                        languages: {
+                          except: %i[id exhibit_id]
+                        },
+                        translations: {
+                          only: %i[locale key value interpolations is_proc]
+                        })
+          ).merge(
+            if_include?(:pages,
+                        searches: { # thumbnail
```

Comment:
Are searches directly associated w/ page exports?  It seems like one might want to be able to export search/browse categories for an exhibit that doesn't really use Feature/About pages and may want to exclude them?  (although perhaps I'm miss-understanding what functionality this bit of code is providing)

Reply:
This matches the previous behavior. My thinking at the time was to avoid weird breakage if you had a page that referenced a browse category, and that browse categories were a type of page (not a `Spotlight::Page`, but an HTML page).